user_auth/models.py

```diff
@@ -0,0 +1,39 @@
+from django.contrib.auth.base_user import BaseUserManager
+from django.contrib.auth.models import AbstractUser
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from django_softdelete.models import SoftDeleteModel
+from model_utils.models import TimeStampedModel
+
+
+class CustomUserManager(BaseUserManager):
+    def create_user(self, email, password, **extra_fields):
+        if not email:
+            raise ValueError(_("The Email must be set"))
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
+        user.save()
+        return user
+
+    def create_superuser(self, email, password, **extra_fields):
+        extra_fields.setdefault("is_staff", True)
+        extra_fields.setdefault("is_superuser", True)
+        extra_fields.setdefault("is_active", True)
+
+        if extra_fields.get("is_staff") is not True:
+            raise ValueError(_("Superuser must have is_staff=True."))
+        if extra_fields.get("is_superuser") is not True:
+            raise ValueError(_("Superuser must have is_superuser=True."))
+        return self.create_user(email, password, **extra_fields)
+
+
+class User(SoftDeleteModel, TimeStampedModel, AbstractUser):
+    username = None
+    email = models.EmailField(_("email address"), unique=True)
```

Comment:
research this
email is unique, but what happens if you delete a user, soft deleted, then you try to create another user with this email ?

Reply:
It won't let me create a new one since the old one is still on the database. We can make a script that hard deletes the user that has been soft deleted for over 60 days for example, or delete another non-unique field and copy the email to it and put the original field to equal the id, I can't change the email to be not unique since logically there should not be two users with the same email.